lib/apiconfig/load.go

```diff
@@ -75,3 +80,23 @@ func LoadClientConfigFromEnvironment() (*CalicoAPIConfig, error) {
 
 	return c, nil
 }
+
+// ClientConfigOverride changes datastore type to Kubernetes when EtcdV3 endpoint is not
+// explicitly defined by user.
+func ClientConfigOverride(cfg *CalicoAPIConfig) (*CalicoAPIConfig, error) {
```

Comment:
I've created a separate function just to be more clear about whats happening. Using `LoadClientConfigFromEnvironment` for everything was a bit confusing for future troubleshooting imo.

Reply:
I would like to see the existing LoadClientConfigFromEnvironment function enhanced to match the current desired behavior, rather than split that logic into two places. While it might be simpler to think about them separately within the context of this change, longer term it will be a lot easier to understand if client configuration loading is done in only one place.